pulpcore/app/models/fields.py

```diff
@@ -35,7 +35,11 @@ def pre_save(self, model_instance, add):
                                'Artifact storage. Files must be stored outside this location '
                                'prior to Artifact creation.'))
         file = super().pre_save(model_instance, add)
-        if file and file._committed and add:
+        if file and file._committed and add and str(file) != upload_to:
+            # the file needs to be moved into place
```

Comment:
This change is a bit questionable in my mind. The problem is that the following lines of code was being executed for uploads that were already in place (via upload_to). The read failed in the case where we're using S3 because of course the file is already stored in S3. So I'm adding a check to see if the file path is equal to upload_to.

Not sure if there's a better way to do this?